lib/common/xml.c

```diff
@@ -551,8 +553,19 @@ pcmk__xe_set_score(xmlNode *target, const char *name, const char *value)
              * value="X+=Y": new value of X is old_value + Y (for some number Y)
              */
             int add = (*v == '+')? 1 : char2score(++v);
+            int new_value_i = pcmk__add_scores(old_value_i, add);
 
-            crm_xml_add_int(target, name, pcmk__add_scores(old_value_i, add));
+            crm_xml_add_int(target, name, new_value_i);
+
+            if (warn) {
+                const char *id = pcmk__xe_id(target);
+
+                crm_warn("Expanded %s='%s' to %s='%d' in <%s%s%s> (behavior is "
+                         "deprecated unless cib_score_update flag is set)",
+                         name, value, name, new_value_i, target->name,
+                         ((id != NULL)? " " PCMK_XA_ID "=" :  ""),
+                         pcmk__s(id, ""));
```

Comment:
This is what I was getting at with:

> One note: The warning from `cibadmin` and `crm_attribute` (via `crm_warn()` in `pcmk__xe_set_score()`) is basically useless, because it appears only when a few `-V` flags are set. Off hand, I'm not sure of a good way around that. One idea is to create a new standard Pacemaker return code for this purpose, to signal that an attribute value was expanded via the deprecated `cib_score_update_warn` flag.
> 
> It's not great form to use an error code to indicate a warning, but hey.

(https://github.com/ClusterLabs/pacemaker/pull/3403#issuecomment-2068820950)

Reply:
It wouldn't be indicating a warning, it would be indicating "success with expansion". The client would choose to make that a warning.